### DISCUSSION TITLE
No SpeedControllers

### DISCUSSION THREAD

**Discussion by nathanmfast** [devcompl/TTController#104] [Q&A] [answered]
This is probably a dumb question, but what happens if you don't specify any SpeedControllers? Does it just go on doing whatever it was doing before? What if I use them and later remove them? Does it go back to whatever it was doing before after having used a service with SpeedControllers and then having removed them? I'm struggling to get the curves right and all I really care about is being able to adjust the LED colors via "normal" methods (long story, but I've got Home Assistant and IOTLink working together to do all kinds of awesome things to my computer as part of my Home Automation setup), which it appears I can do by swapping out config files and then restarting the Windows Service. It'd be cool if this was more like an API but I'm extremely grateful that this exists because the TT RGB Plus software has made me want to stab myself in the eye for more than a year and this is finally something that works reliably!

**Comment by MoshiMoshi0** [comment, marked as answer]
If you stop sending fan speed commands it will just use the last set speed. You should probably setup a boot profile to save the default speed to the controller.

> It'd be cool if this was more like an API

You can use IPC plugins to send colors via websocket:
https://moshimoshi0.github.io/TTController/plugins/effects/ipceffect

**Reply by nathanmfast** [reply]
That's perfect. You're awesome!
